sqlmesh/core/context.py

```diff
@@ -756,52 +756,58 @@ def table_diff(
         self,
         source: str,
         target: str,
+        model_or_snapshot: ModelOrSnapshot,
```

Comment:
The API simplification is predicated on tightening the input. We expect source and target to be environment names only, _not_ snapshot names. `model_or_snapshot` should really just be a model name.

While it might be less flexible, it seems far more likely a user will diff a model across environments than a user will diff 2 snapshots. They will not be able to easily obtain these snapshots ids. Comparing an orphaned snapshot (IE a snapshot not connected to an environment) probably would not work, therefore you get the same utility this way since it should always be relative to an environment. 
The difference is now the input/API is uniform.

Reply:
Reverted, we can managed both so long as the simplified interface is intact
`sqlmesh table_diff dev:prod some.table`

will work just as well as 
`sqlmesh table_diff sushi__alex.top_waiters:sushi__toby.top_waiters -O waiter_id`

will work just as well as 
`sqlmesh table_diff sqlmesh__sushi.sushi__top_waiters__1307081719:sqlmesh__sushi.sushi__top_waiters__477763947 -O waiter_id`


Only in the first case is `grain` identifiable and should be the preferred pattern.